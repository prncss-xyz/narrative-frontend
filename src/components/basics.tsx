import styled from "@emotion/styled";
import {
  border,
  color,
  flexbox,
  grid,
  layout,
  space,
  system,
  typography,
} from "styled-system";

const custom = system({
  gap: { property: "gap", scale: "space" },
  columnGap: { property: "columnGap", scale: "space" },
  left: { property: "left", scale: "space" },
  top: { property: "top", scale: "space" },
  textDecoration: true,
  cursor: true,
  position: true,
});

export const Img = styled.img(
  {
    boxSizing: "border-box",
    minWidth: 0,
  },
  border,
  space,
  layout
);

export const Box = styled.div(
  {
    boxSizing: "border-box",
    minWidth: 0,
  },
  border,
  space,
  layout,
  color,
  flexbox,
  grid,
  typography,
  custom
);

export const Flex = styled(Box)({ display: "flex" });

export const Grid = styled(Box)({ display: "grid" });

export const InputRaw = styled.input(
  {
    boxSizing: "border-box",
    minWidth: 0,
  },
  border,
  space,
  layout,
  color
);

export function H1({
  children,
  ...props
}: {
  children: React.ReactNode;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box
      as="h1"
      my={4}
      textAlign="center"
      fontWeight="normal"
      fontSize={5}
      {...props}
    >
      {children}
    </Box>
  );
}

export function H2({
  children,
  ...props
}: {
  children: React.ReactNode;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box
      as="h2"
      my={4}
      textAlign="center"
      fontWeight="normal"
      fontSize={4}
      {...props}
    >
      {children}
    </Box>
  );
}

export function H3({
  children,
  ...props
}: {
  children: React.ReactNode;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box
      as="h3"
      my={1}
      color="muted"
      fontWeight="normal"
      textDecoration="underline"
      textAlign="left"
      fontSize={1}
      {...props}
    >
      {children}
    </Box>
  );
}

export function ActionBox({
  children,
  ...props
}: {
  children: React.ReactNode;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box my={1} color="muted" backgroundColor="bgLight" py={1} px={2} {...props}>
      {children}
    </Box>
  );
}

export function OverlayActionBox({
  children,
  ...props
}: {
  children: React.ReactNode;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box my={1} color="muted" backgroundColor="accent" py={1} px={2} {...props}>
      {children}
    </Box>
  );
}
