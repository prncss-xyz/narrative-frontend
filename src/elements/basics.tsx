import styled from "@emotion/styled";
import {
  border,
  color,
  flexbox,
  grid,
  layout,
  space,
  system,
  typography,
} from "styled-system";

const custom = system({
  gap: { property: "gap", scale: "space" },
  left: { property: "left", scale: "space" },
  top: { property: "top", scale: "space" },
  textDecoration: true,
  cursor: true,
  position: true,
});

export const Img = styled.img(border, space, layout);

export const Box = styled.div(
  {
    boxSizing: "border-box",
    minWidth: 0,
  },
  border,
  space,
  layout,
  color,
  flexbox,
  grid,
  typography,
  custom
);

export const Flex = styled(Box)({ display: "flex" });

export const Grid = styled(Box)({ display: "grid" });

export function H1({
  children,
  ...props
}: {
  children: React.ReactNode;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box
      as="h1"
      my={4}
      textAlign="center"
      fontWeight="normal"
      fontSize={5}
      {...props}
    >
      {children}
    </Box>
  );
}

export function H2({
  children,
  ...props
}: {
  children: React.ReactNode;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box
      as="h2"
      my={4}
      textAlign="center"
      fontWeight="normal"
      fontSize={4}
      {...props}
    >
      {children}
    </Box>
  );
}

export function H3({
  children,
  ...props
}: {
  children: React.ReactNode;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box
      as="h3"
      my={1}
      color="tone4"
      fontWeight="normal"
      textDecoration="underline"
      fontSize={2}
      {...props}
    >
      {children}
    </Box>
  );
}

export function ActionBox({
  children,
  ...props
}: {
  children: React.ReactNode;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box my={1} color="tone4" backgroundColor="tone1" py={1} px={2} {...props}>
      {children}
    </Box>
  );
}

export function OverlayActionBox({
  children,
  ...props
}: {
  children: React.ReactNode;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box my={1} color="tone4" backgroundColor="tone0" py={1} px={2} {...props}>
      {children}
    </Box>
  );
}
