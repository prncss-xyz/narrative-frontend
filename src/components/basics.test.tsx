import { render } from "@testing-library/react";
import { ActionBox, Box, Flex, H1, H3, Img, OverlayActionBox } from "./basics";

describe("basics", () => {
  describe("Img", () => {
    it("should render properly", () => {
      const { container } = render(<Img />);
      expect(container).toMatchSnapshot();
    });
  });
  describe("Box", () => {
    it("should render properly", () => {
      const { container } = render(<Box />);
      expect(container).toMatchSnapshot();
    });
  });
  describe("Flex", () => {
    it("should render properly", () => {
      const { container } = render(<Flex />);
      expect(container).toMatchSnapshot();
    });
  });
  describe("H1", () => {
    it("should render properly", () => {
      const { container } = render(<H1>H1</H1>);
      expect(container).toMatchSnapshot();
    });
  });
  describe("H2", () => {
    it("should render properly", () => {
      const { container } = render(<H3>H2</H3>);
      expect(container).toMatchSnapshot();
    });
  });
  describe("H3", () => {
    it("should render properly", () => {
      const { container } = render(<H3>h3</H3>);
      expect(container).toMatchSnapshot();
    });
  });
  describe("ActionBox", () => {
    it("should render properly", () => {
      const { container } = render(<ActionBox>h3</ActionBox>);
      expect(container).toMatchSnapshot();
    });
  });
  describe("OverlayActionBox", () => {
    it("should render properly", () => {
      const { container } = render(<OverlayActionBox>h3</OverlayActionBox>);
      expect(container).toMatchSnapshot();
    });
  });
});
