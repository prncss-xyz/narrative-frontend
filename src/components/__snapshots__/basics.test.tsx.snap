// Vitest Snapshot v1

exports[`basics > ActionBox > should match snapshot 1`] = `<div />`;

exports[`basics > Box > should match snapshot 1`] = `<div />`;

exports[`basics > Flex > should match snapshot 1`] = `<div />`;

exports[`basics > H1 > should match snapshot 1`] = `<div />`;

exports[`basics > H2 > should match snapshot 1`] = `<div />`;

exports[`basics > H3 > should match snapshot 1`] = `<div />`;

exports[`basics > Img > should match snapshot 1`] = `
<div>
  <img
    class="css-qdb0d0-Img e2i8k6n4"
  />
</div>
`;

exports[`basics > OverlayActionBox > should match snapshot 1`] = `<div />`;
